x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_host_count.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as rt from 'io-ts';
+import { ES_SEARCH_STRATEGY, IKibanaSearchResponse } from '@kbn/data-plugin/common';
+import { useCallback, useEffect } from 'react';
+import { catchError, map, Observable, of } from 'rxjs';
+import createContainer from 'constate';
+import { estypes } from '@elastic/elasticsearch';
+import { decodeOrThrow } from '../../../../../common/runtime_types';
+import { useDataSearch, useLatestPartialDataSearchResponse } from '../../../../utils/data_search';
+import { useMetricsDataViewContext } from './use_data_view';
+import { useUnifiedSearchContext } from './use_unified_search';
+
+export const useHostCount = () => {
```

Comment:
We don't need a new API for this. I decided to go with `data` plugin directly

Reply:
Do we need to add it as a dependency in the config?